__tests__/wallet/walletServiceStorageProxy.test.ts

```diff
@@ -0,0 +1,565 @@
+        .mockResolvedValueOnce(mockTx1Response)
+        .mockResolvedValueOnce(mockTx2Response);
+
+      // Convert to async generator result array
+      const results = [];
+      for await (const item of proxiedStorage.getSpentTxs(mockInputs)) {
+        results.push(item);
+      }
+
+      expect(results).toHaveLength(3);
+      expect(results[0].input).toBe(mockInputs[0]);
+      expect(results[0].index).toBe(0);
+      expect(results[0].tx).toBeDefined();
+      expect(results[0].tx.tx_id).toBe('tx1');
+      expect(results[1].input).toBe(mockInputs[1]);
+      expect(results[1].index).toBe(1);
+      expect(results[1].tx.tx_id).toBe('tx2');
+      expect(results[2].input).toBe(mockInputs[2]);
+      expect(results[2].index).toBe(2);
+      expect(results[2].tx.tx_id).toBe('tx1'); // Should use cached tx1
+    });
+
+    it('should cache transactions to avoid duplicate fetches', async () => {
+      const mockTx1Response = {
+        success: true,
+        tx: {
+          hash: 'tx1',
+          version: 1,
+          weight: 5.0,
+          timestamp: 1000000,
+          inputs: [],
+          outputs: [],
+          parents: [],
+          tokens: [],
+        },
+        meta: {
+          height: 500,
+          first_block: 'first',
+          voided_by: [],
+        },
+      };
+
+      const mockTx2Response = {
+        success: true,
+        tx: {
+          hash: 'tx2',
+          version: 1,
+          weight: 5.0,
+          timestamp: 2000000,
+          inputs: [],
+          outputs: [],
+          parents: [],
+          tokens: [],
+        },
+        meta: {
+          height: 600,
+          first_block: 'second',
+          voided_by: [],
+        },
+      };
+
+      (wallet.getFullTxById as jest.Mock)
+        .mockResolvedValueOnce(mockTx1Response)
+        .mockResolvedValueOnce(mockTx2Response);
+
+      const results = [];
+      for await (const item of proxiedStorage.getSpentTxs(mockInputs)) {
+        results.push(item);
+      }
+
+      // Should only call getFullTxById twice (once for each unique tx)
+      expect(wallet.getFullTxById).toHaveBeenCalledTimes(2);
+      expect(wallet.getFullTxById).toHaveBeenCalledWith('tx1');
+      expect(wallet.getFullTxById).toHaveBeenCalledWith('tx2');
+      expect(results).toHaveLength(3);
+    });
+
+    it('should skip inputs when transaction fetch fails', async () => {
+      const mockTx1Response = {
+        success: true,
+        tx: {
+          hash: 'tx1',
+          version: 1,
+          weight: 5.0,
+          timestamp: 1000000,
+          inputs: [],
+          outputs: [],
+          parents: [],
+          tokens: [],
+        },
+        meta: {
+          height: 500,
+          first_block: 'first',
+          voided_by: [],
+        },
+      };
+
+      (wallet.getFullTxById as jest.Mock)
+        .mockResolvedValueOnce(mockTx1Response)
+        .mockRejectedValueOnce(new Error('Transaction not found'));
```

Comment:
question: Should we really handle a tx not found by silently ignoring it? Is this a good developer experience in this situation?